pkg/providers/launchtemplate/launchtemplate.go

```diff
@@ -424,3 +431,26 @@ func (p *Provider) DeleteLaunchTemplates(ctx context.Context, nodeClass *v1beta1
 	}
 	return nil
 }
+
+func (p *Provider) ResolveClusterCIDR(ctx context.Context) error {
+	if p.ClusterCIDR != nil {
+		return nil
+	}
+	out, err := p.eksapi.DescribeClusterWithContext(ctx, &eks.DescribeClusterInput{
+		Name: aws.String(options.FromContext(ctx).ClusterName),
+	})
+	if err != nil {
+		return fmt.Errorf("failed to resolve cluster CIDR, %w", err)
+	}
+	if ipv4CIDR := out.Cluster.KubernetesNetworkConfig.ServiceIpv4Cidr; ipv4CIDR != nil {
+		p.ClusterCIDR = ipv4CIDR
+		logging.FromContext(ctx).Debugf("discovered cluster CIDR %q", p.ClusterCIDR)
```

Comment:
```suggestion
		logging.FromContext(ctx).With("cluster-cidr", p.ClusterCIDR).Debugf("discovered cluster CIDR")
```

nit: consider using `With()` since this is a structured value